tests/baselines/reference/keyofAndIndexedAccess.types

```diff
@@ -125,7 +125,7 @@ type NAME = "name";
 >NAME : "name"
 
 type WIDTH_OR_HEIGHT = "width" | "height";
->WIDTH_OR_HEIGHT : "width" | "height"
+>WIDTH_OR_HEIGHT : WIDTH_OR_HEIGHT

```

Comment:
This happens because we now call `getTypeFromTypeNode` in `checkUnionOrIntersectionType`. We used to not do that, which means we wouldn't report errors on excessively large unions unless you referenced them somewhere. That was inconsistent.